python/hsfs/engine/spark.py

```diff
@@ -53,10 +53,48 @@ def __init__(self):
         self._spark_session.conf.set("hive.exec.dynamic.partition.mode", "nonstrict")
         self._spark_session.conf.set("spark.sql.hive.convertMetastoreParquet", "false")
 
+        self.validate_spark_configuration()
+
         if importlib.util.find_spec("pydoop"):
             # If we are on Databricks don't setup Pydoop as it's not available and cannot be easily installed.
             util.setup_pydoop()
 
+    def validate_spark_configuration(self):
+        key = "spark.hadoop.hops.ssl.trustore.name"
+        if(not self._spark_session.conf.get(key)):
+            raise FeatureStoreException("Invalid: " + key)
+        key = "spark.hadoop.hops.rpc.socket.factory.class.default"
+        if(not self._spark_session.conf.get(key) == "io.hops.hadoop.shaded.org.apache.hadoop.net.HopsSSLSocketFactory"):
```

Comment:
I am not sure what the data type of `conf` actually is, but I would think it is a `dict` so are you sure it doesn't throw a `KeyError` when the setting doesn't exist at all? This might need to be handled for properties which are our own, and are not available at all on other environments.